src/squlearn/encoding_circuit/qcnn_encoding_circuit.py

```diff
@@ -0,0 +1,577 @@
+        self._operations_list = []  # stores the operations applied in the qcnn
+        self._default = default
+        if default:
+            if num_qubits == 0:
+                print("To generate a default circuit provide a number of qubits > 0.")
+            else:
+                self.default_circuit()
+
+    @property
+    def num_parameters(self) -> int:
+        """Returns the number of trainable parameters of the current encoding circuit."""
+        return self._num_parameters
+
+    @property
+    def left_qubits(self) -> list:
+        """Returns the qubits which one can operate on in the current circuit as list."""
+        return self._left_qubits
+
+    @property
+    def operations_list(self) -> list:
+        """Returns the list of operators currently acting on the encoding circuit."""
+        return self._operations_list
+
+    def set_num_qubits(self, n: int = 0):
+        """Set the number of qubits of the circuit."""
+        if n > 0:
+            if self.num_qubits == n:
+                print("The number of qubits did not change.")
+            else:
+                self._num_qubits = n
+                self._left_qubits = [i for i in range(n)]
+                self._num_parameters = 0
+                if self._default:
+                    self._operations_list = []
+                    self.default_circuit()
+                else:
+                    for operation in self.operations_list:
+                        if operation[0] == "C":
+                            self.convolution(*operation[1:], new_operation=False)
+                        elif operation[0] == "P":
+                            self.pooling(*operation[1:], new_operation=False)
+                        elif operation[0] == "F":
+                            self.fully_connected(*operation[1:], new_operation=False)
+                            break
+
+    def set_num_features(self, n: int = 0):
+        """Change the number of features in the feature map of the circuit."""
+        if n > 0:
+            self._num_features = n
+
+    def convolution(
+        self,
+        QC="None",
+        label: str = "C",
+        alternating: bool = True,
+        diff_params: bool = True,
+        new_operation: bool = True,
+    ):
+        """
+        Add a convolution layer to the encoding circuit.
+
+        Args:
+            QC Union[EncodingCircuitBase,QuantumCircuit]: The quantum circuit, which is applied on every qubit
+                modulo qubits of this circuit.
+            label (str): Sets the name of the operation.
+            alternating (bool): It applies the QC on every qubit modulo qubits of this circuit beginning at 0
+                and if True it applies the QC on every qubit beginning at 1 again.
+            diff_params (bool): If True, different parameters are used for the gates build by this layer.
+        """
+        # Define default circuit
+        if QC == "None":
```

Comment:
```suggestion
        if not quantum_circuit:
```
Using `None` also allows for a more readable if statement here